_posts/2018-01-19-config-parsing-order.md

```diff
@@ -0,0 +1,308 @@
+
+Everywhere else, the last mention of a single-valued option has “priority” and overrides a previous value. The same will now be true of `--config` expansion. Like other expansion options, `--config` will now expand to its rc-defined expansion “in-place,” so that the options it expands to have the same precedence. 
+
+Since this is no longer a fixed-point expansion, there are a few other changes:
+
++  `--config=foo --config=foo` will be expanded twice. If this is undesirable, more care will need to be taken to avoid redundancy. Double occurences will cause a warning.
++  cycles are no longer implicitly ignored, but will error out.
+
+Other rc ordering semantics remain. "common" options are expanded first, followed by the command hierarchy. This means that for an option added on the line "`build:foo --buildopt`", it will get added to `--config=foo`'s expansion for bazel build, test, coverage, etc. "`test:foo --testopt`" will add `--testopt` after the (less specific and therefore lower priority) build expansion of `--config=foo`. If this is confusing, avoid alternating command types in the rc file, and group them in order, general options at the top. This way, the order of the file is close to the interpretation order.
+
+## How to start using new behavior 
+
+1. Check your usual `--config` values’ expansions by running your usual bazel command line with `--announce_rc`. The order that the configs are listed, with the options they expand to, is the order in which they are interpreted.
+
+1. Spend some time understanding the applicable configs, and check if any configs expand to the same option. If they do, you may need to move rc lines around to make sure the same value has priority with the new ordering.  See “Suggestions for config definers.”
+
+1. Flip on the startup option `--expand_configs_in_place` and debug any differences using `--announce_rc` 
+
+   _If you have a shared bazelrc for your project, note that changing it will propagate to other users who might be importing this bazelrc into a personal rc. Proceed with caution as needed_
+
+1. Add the startup option to your bazelrc to continue using this new expansion order.
+
+### Suggestions for config definers
+
+You might be in a situation where you own some `--config` definitions that are shared between different people, even different teams, so it might be that the users of your config are using both `--expand_configs_in_place` behavior and the old, default behavior.
+
+In order to minimize differences between old and new behavior, here are some tips. 
+
+1. Avoid internal conflicts within a config expansion (redefinitions of the same option)
+1. Define recursive `--config` at the END of the config expansion
+   - Only critical if #1 can’t be followed.
+
+#1 is especially important if the config expands to another config. The behavior will be more predictable with `--expand_configs_in_place`, but without it, the expansion of a single `--config` depends on previous `--configs`. 
```

Comment:
This line will render as `<h1>` because it starts with "#". I think you can escape it, or just rephrase the sentence, e.g. "Point #1 above is..."